### ISSUE TITLE
Bug: `ConfigError: Config (unnamed): Key "plugins": Cannot redefine plugin "@".` when using `ESLint` class API with `overrideConfigFile: true`

### ISSUE THREAD

**Issue by JounQin** [eslint/eslint#19722]
### Environment

Node version: v20.19.1
npm version: v10.8.2
Local ESLint version: v9.26.0 (Currently used)
Global ESLint version: Not found
Operating System: darwin 24.4.0

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>API usage</summary>

```js
await eslint.lintText(text, {
  filePath,
  warnIgnored: true,
})
```
</details>

### What did you expect to happen?

No error without [`delete eslintConfig.overrideConfig.plugins;`](https://github.com/prettier/prettier-eslint/pull/1190/files#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80R250)

### What actually happened?

```log
ConfigError: Config (unnamed): Key "plugins": Cannot redefine plugin "@".
```

### Link to Minimal Reproducible Example

https://github.com/prettier/prettier-eslint/pull/1190

Reproduce steps:

```sh
yarn install
# remove `delete eslintConfig.overrideConfig.plugins;` at src/index.ts#L250
yarn format
```

### Participation

- [x] I am willing to submit a pull request for this issue.

### Additional comments

I'm adding `eslint` v9 support for https://github.com/prettier/prettier-eslint at https://github.com/prettier/prettier-eslint/pull/1190
